src/mlpack/methods/sparse_svm/sparse_svm_function_impl.hpp

```diff
@@ -36,44 +42,124 @@ void SparseSVMFunction::Shuffle()
   labels = std::move(newLabels);
 }
 
+double SparseSVMFunction::Evaluate(const arma::mat& parameters)
+{
+  arma::mat params = parameters.head_cols(parameters.n_cols - 1);
```

Comment:
This actually will perform a copy, so it would be faster to use `parameters.head_cols().t()` directly in the expression of `dots`.